apps/datafeeder/src/app/presentation/components/upload-data-rules/upload-data-rules.component.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing'
+
+import { UploadDataRulesComponent } from './upload-data-rules.component'
+
+describe('UploadDataRulesComponent', () => {
+  let component: UploadDataRulesComponent
+  let fixture: ComponentFixture<UploadDataRulesComponent>
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [UploadDataRulesComponent],
+    }).compileComponents()
+  }))
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UploadDataRulesComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
```

Comment:
Ok now you should thinking of writing more tests.
In this component, I want to test that depending on the input fromat, I have the correct layout (number of lines etc...)

Use something like 
```js
const el = fixture.debugElement.query(By.css('.label')).nativeElement
``` 
to inspect the HTML, check other tests for inspiration.